sbpy/photometry/core.py

```diff
@@ -551,52 +573,142 @@ def phaseint(self):
         """Phase integral"""
         return self._phase_integral()
 
-    def fit(self, eph):
-        """Fit photometric model to photometric data stored in sbpy.data.Ephem
-        object
+    def fit(self, eph, mag, fitter=None, return_fitter=False, **kwargs):
```

Comment:
I believe so, because uncertainty is automatically supported by the `weights` keyword of almost all fitter classes.  But I will need to test it and also add this into the docs.

Reply:
Yes we can use `weights` keyword for all the fitter classes currently defined in `astropy.modeling.fitting` submodule.  I will add this note to the docstring.